debug/debug.go

```diff
@@ -0,0 +1,36 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug // import "go.opentelemetry.io/otel/debug"
```

Comment:
Could they, yes.  I opted for a package of some kind to hide the API from the user which should only need this if something is going wrong.  Do you have a suggestion on what might be a more appropriate name?

Reply:
Well if we go and instrument our API or  SDK with this there would be no way to have this as a module, and optionally include it.

What would be a good name for the setting of this to convey what it does?  I'm concerned that going with `otel.SetLogger` or even `otel.SetInternalLogger` isn't enough to convey that this doesn't expose the OpenTelemetry (non-existant) API for logging.  Keeping it separate, by a namespace of some kind, is a signal that this does something different from the SetTracerProvider etc.

Reply:
> What would be a good name for the setting of this to convey what it does? I'm concerned that going with otel.SetLogger or even otel.SetInternalLogger isn't enough to convey that this doesn't expose the OpenTelemetry (non-existant) API for logging. Keeping it separate, by a namespace of some kind, is a signal that this does something different from the SetTracerProvider etc.

Yeah, that is a good problem to recognize. Honestly, I'm not sure outside of a good name there is going to be a way to communicate what your are setting without being able to rely on the reader reading the Go docs. Even if it is in its own package.

Would it be worth considering names like `MsgHandler` or `MessageHandler`?

Reply:
In the current iteration it is `otel.SetLogger()`.  While I'm not happy with it, I am willing to live with it, only because we will most likely call any logging API something akin to `SetLoggerProvider`.

Are we willing to live with this, or would you like to see an iteration with MessageHandler?

Reply:
I have a slight preference for `otel.SetInternalLogger()`, but I think either is workable.